linkerd/app/outbound/src/lib.rs

```diff
@@ -30,7 +30,7 @@ use std::{
     time::Duration,
 };
 
-mod discover;
+pub mod discover;
```

Comment:
This isn't exactly a blocker but it gives me pause. What outbound-specific discovery logic are we trying to expose?

Reply:
this was made public as the `gateway` crate also needs to construct the `WithAllowlist` type with its own allowlist. moving that code into `linkerd-proxy-service-profiles` will obviate the need for that!